samples/matter/common/src/bridge/bridge_manager.h

```diff
@@ -6,35 +6,56 @@
 
 #pragma once
 
+#include "bridge_util.h"
 #include "bridged_device.h"
 #include "bridged_device_data_provider.h"
-#include <map>
 
 class BridgeManager {
 public:
 	void Init();
-	CHIP_ERROR AddBridgedDevice(BridgedDevice::DeviceType bridgedDeviceType, const char *nodeLabel);
+	CHIP_ERROR AddBridgedDevices(BridgedDevice *aDevice, BridgedDeviceDataProvider *aDataProvider);
 	CHIP_ERROR RemoveBridgedDevice(uint16_t endpoint);
-	static BridgedDevice *GetBridgedDevice(uint16_t endpoint, const EmberAfAttributeMetadata *attributeMetadata,
-					       uint8_t *buffer);
-	static CHIP_ERROR HandleRead(uint16_t endpoint, chip::ClusterId clusterId,
+	static CHIP_ERROR HandleRead(uint16_t index, chip::ClusterId clusterId,
 				     const EmberAfAttributeMetadata *attributeMetadata, uint8_t *buffer,
 				     uint16_t maxReadLength);
-	static CHIP_ERROR HandleWrite(uint16_t endpoint, chip::ClusterId clusterId,
+	static CHIP_ERROR HandleWrite(uint16_t index, chip::ClusterId clusterId,
 				      const EmberAfAttributeMetadata *attributeMetadata, uint8_t *buffer);
 	static void HandleUpdate(BridgedDeviceDataProvider &dataProvider, chip::ClusterId clusterId,
 				 chip::AttributeId attributeId, void *data, size_t dataSize);
 
 private:
-	CHIP_ERROR AddDevice(BridgedDevice *device);
-
 	static constexpr uint8_t kMaxBridgedDevices = CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
 	static constexpr uint8_t kMaxDataProviders = CONFIG_BRIDGE_MAX_BRIDGED_DEVICES_NUMBER;
```

Comment:
It seems it is not used anywhere. For now we create a new data provider for every bridged device, but it doesn't need to be this way always. I think that we should somehow validate that we will not create more providers than it's assumed.

Reply:
Ok, I added a check for that in the BridgeManager.